src/application/SafeService.ts

```diff
@@ -68,9 +68,16 @@ export default class SafeService implements ISafeService {
         'Missing Drip List ID while linking pending voting round.',
       );
 
-    const safeTx = await this.getSafeTransaction(safeTransactionHash);
+    const { isExecuted, isSuccessful, safeAddress } =
+      await this.getSafeTransaction(safeTransactionHash);
 
-    if (safeTx.isExecuted) {
+    if (safeAddress !== votingRound.publisherAddress) {
+      shouldNeverHappen(
```

Comment:
A note on `shouldNeverHappen` errors:

These are errors that should never occur when the code is correct. If they are thrown, it means there's a bug _elsewhere_. The "entry component" should have caught it. It's a way to identify those.